### DISCUSSION TITLE
Awesome Feature Ideas!

### DISCUSSION THREAD

**Discussion by EYHN** [toeverything/AFFiNE#6180] [Ideas]
# General

* https://github.com/toeverything/AFFiNE/discussions/6186
* https://github.com/toeverything/AFFiNE/discussions/6256

# Journal

* https://github.com/toeverything/AFFiNE/discussions/6175
* https://github.com/toeverything/AFFiNE/discussions/6184

# Editor

* https://github.com/toeverything/AFFiNE/discussions/6172
* https://github.com/toeverything/AFFiNE/discussions/6173
* https://github.com/toeverything/AFFiNE/discussions/6178
* https://github.com/toeverything/AFFiNE/discussions/6177
* https://github.com/toeverything/AFFiNE/discussions/6179
* https://github.com/toeverything/AFFiNE/discussions/6149
* https://github.com/toeverything/AFFiNE/discussions/6148
* https://github.com/toeverything/AFFiNE/discussions/6189
* https://github.com/toeverything/AFFiNE/discussions/6290
* https://github.com/toeverything/AFFiNE/discussions/6346

# Third-party integrations

* https://github.com/toeverything/AFFiNE/discussions/6174
* https://github.com/toeverything/AFFiNE/discussions/6176

# Other tools

### Browser Extension

* https://github.com/toeverything/AFFiNE/discussions/6171

**Comment by lawvs** [comment]
I suggest adding an automatic code language detection feature for code blocks :)

**Reply by EYHN** [reply]
https://github.com/toeverything/AFFiNE/discussions/6189

**Comment by Aninhanesuto** [comment]
That's great

**Comment by Zhaith-Izaliel** [comment]
Hey! I really love AFFiNE and the work the team put into that awesome app. 

I would really like to see a classical directory tree for notes. Since it can be used as a knowledge base, having the possibility to create a standard documentation in a workspace with organized notes would be awesome. Especially since using the tag and collection systems aren't always covering all my use cases.

I don't know if it was already discussed and discarded, or planned. But it would be one of the main feature I would like to see. And I don't think I'm the only one :)

**Reply by houshasen** [reply]
Like this request? https://github.com/toeverything/AFFiNE/discussions/6988

**Reply by Zhaith-Izaliel** [reply]
Yep exactly, glad to see it's already been worked on :)

**Comment by LYUYork** [comment]
embed blocks like picture,link in "table" and "mind map" #9931

Graph: add link to other docs #9936

**Comment by usshauler** [comment]
I love Affine and it have dropped all other note taking apps for this one. However, there are still a few features I wish were implemented here and these are:

1. Share > Export to clipboard (basically copies everything on your page to clipboard including pictures to easily paste elsewhere. Only app that I have seen do this is Appflowy.
2. Add lines and spacing view within the Code Blocks - Think VSCode view of YAML files or code within the lines dropping in the backend to show you spacing. Have a setting to enable and disable this please.
3. Sort all your documents in alphabetical order within a folder or from your main Workspace. I like my organization and need this in my life for this app.
4. Add some form of encryption to your sync server. I know this would probably take some time to do but it would be nice as an added security benefit overall for everyone and peace of mind.

Thank you for everything Affine Team !

**Reply by dvalnn** [reply]
Export markdown clipboard would be an amazing feature, especially if coupled with the reverse import from clipboard!

I have been feeling the need for both of these features lately and end up having to create intermediate buffer files for copy-pasting around.

**Comment by smosgasbord** [comment]
Expose the API or webhook so that we can also interact with other automation services that will help Affine see traction as a real Notion alternative. We need to be able to connect it to N8N or make or zapier easily, we can do http requests to generate pages or receive events like a change on a table.

please please.

**Comment by ddean6232** [comment]
API Please!

**Reply by almereyda** [reply]
Affine uses a GraphQL backend that you could try to hook into.

I'm not able to set a personal API key from my account settings, though.

Could be interesting to see if authenticating against a configured OAuth IdP would suffice to authorize requests against that GraphQL endpoint similar to how the app does.

**Comment by almereyda** [comment]
Please also consider calendar views on inline databases. That would render the feature pretty much complete.

- https://github.com/toeverything/AFFiNE/discussions/12992

With respective data present (beginning and end fields), also a GANTT view seems useful.

**Comment by bertyleamon** [comment]
Hello Affine team! I'm a new user, trying it out after long-time use of alternatives like notion, obsidian, capacities, anytype, and others. 

One feature I'd like to see added is the ability to create a template that can be automatically applied to new daily journal entries. This would help reduce friction for structuring my daily journals so I can easily keep track of all parts of my routine. 

Currently, users have to:
1) make a page for the template [ie daily journal template]
2) duplicate this template page to create a new note each day
3) open properties in the duplicated note to apply it to the current date

Because opening the journal creates a blank page by default, this current method often requires users to delete a blank page to avoid two notes being assigned to the same date in the journal.

When I asked affine AI about this, here is what it said:

***

That's a fantastic idea! While AFFiNE doesn't currently have a feature to automatically apply a chosen template every day, you can still streamline the process to make it very quick and easy to use your preferred journal template daily.
Here's the most efficient way to set it up:

**Create Your Master Journal Template:**
Create a new page in AFFiNE and design it exactly how you want your daily journal to look. Include all the headings, lists, Kanban boards, or any other blocks you use regularly. Give it a clear title like "Daily Journal Template" or similar.

**Duplicate for Daily Use:**
When you're ready to start a new daily journal entry, go to your "Daily Journal Template" page.
Use the "Duplicate" function (usually found in the page options or context menu) to create a copy of this template.
Rename the duplicated page to the current date (e.g., "March 26, 2026 Journal").

Crucially, link it to the current date: Within the new daily journal page, type @ followed by the current date (e.g., @March 26, 2026). This will ensure your entry appears in the "Journals" view, helping you keep track of all your daily entries in one place  .

By duplicating your master template and then linking it to the current date, you get a fresh, pre-structured journal entry ready for your thoughts without having to build it from scratch every time. It's a quick two-step process that feels almost automatic once you get used to it!

***

I hope this can be added as a feature to streamline this process in the future. Thank you for considering it and for all your work in developing affine!

**Reply by bertyleamon** [reply]
Update: I found a workaround in "settings > preferences > my templates > template for journal". Once I made a template doc and checked the template box in its properties, I was able to select it from the dropdown menu in the settings as noted. It wasn't intuitive to find, and was not suggested by affine AI when I asked if giving journals a template by default was an option. When I did find this setting, it was a pleasant surprise - well done! Could be smoother, but I'm glad it actually is possible,

**Comment by turboslon** [comment]
I would very much like to be able to save the view settings per collection/tag (see #13433)

**Comment by Senstei** [comment]
Hi there, there is a smal feature that would be so helpfull for me, espacially when I use the locally hosted Desktop App: 

The trash needs a "select all" button like the there is one at the page "all documents". An "empty trash" button would also do. Right now I've to select each document separately to permamently delete it. After I  tried to import my notion workspace and needed to redo the process, I've currently 5000 documents in the locally stored trash....

Thank you very much

**Reply by lawvs** [reply]
I've opened https://github.com/toeverything/AFFiNE/pull/15556/ to address this issue. Thanks for the feedback!
